packages/amplify-cli/src/extensions/amplify-helpers/push-resources.ts

```diff
@@ -78,8 +78,8 @@ export const pushResources = async (
 
   // no changes detected
   if (!hasChanges && !context.exeInfo.forcePush && !rebuild) {
+    context.usageData.stopCodePathTimer(ManuallyTimedCodePath.PUSH_TRANSFORM);
     printer.info('\nNo changes detected');
-
```

Comment:
Should this go after the print? Not sure if it really makes a difference in the metric, but it might be worth factoring in.